docs/docs/grpc-server.md

```diff
@@ -0,0 +1,85 @@
+---
+layout: docs
+title: gRPC servers
+permalink: grpc/server/
+---
+
+# gRPC servers
+
+Mu-Haskell defines a generic notion of service and server that implements it. This generic server can then be used by `mu-grpc-server`, to provide a concrete implementation using a specific wire format.
+
+## Implementing the service
+
+Let's get back to the example we used in the [generic RPC section]({% link docs/rpc.md %}). In order to implement the corresponding service, you have to define the behavior of each method by means of a *handler*. You can use Haskell types for your handlers, given that you had previously declared that they can be mapped back and forth the schema types using `ToSchema` and `FromSchema`. For example, the following is a handler for the `SayHello` method in `Greeter`:
+
+```haskell
+sayHello :: (MonadServer m) => HelloRequest -> m HelloResponse
+sayHello (HelloRequest nm) = pure $ HelloResponse ("hi, " <> nm)
+```
+
+Notice the use of `MonadServer` in this case. This gives us the ability to:
+
+* Run arbitrary `IO` actions by using `liftIO`,
+* Return an error code by calling `serverError`.
+
+Being polymorphic here allows us to run the same server in multiple back-ends. Furthermore, by enlarging the set of abilities required for our monad `m`, we can [integrate with other libraries](transformer.md), including logging and resource pools.
```

Comment:
Will this link to `transformer.md` work right away?

Reply:
Nope, we need to use the special syntax.

Reply:
Thanks for the catch!